mmdet/models/detectors/base.py

```diff
@@ -213,29 +213,61 @@ def forward(self, img, img_metas, return_loss=True, **kwargs):
         else:
             return self.forward_test(img, img_metas, **kwargs)
 
-    def forward_export(self, imgs):
+    def forward_export(self, imgs, img_metas=None):
+        """
+        This function is called from the function `forward_test` in one of the cases:
+        * During export of the model. In this case
+          * `forward_export_context` is used,
+          * `img_metas` is passed to the model using the field `self.img_metas`
+        * During compression of the model by NNCF, for building graph of
+          the model using `dummy_forward` function. In this case
+          * `forward_export_context` is used too,
+          * also `img_metas` is passed to the model using the field `self.img_metas`
+        * During compression of the model by NNCF, for initialization of
+          the model. In this case
+          * `forward_nncf_initialization_context` is used
+          * `img_metas` is passed to the model through parameter of this function.
```

Comment:
Why does the same flow as for export does not work for NNCF initialization?

Reply:
Pavel, the same flow cannot be used for NNCF initialization, since during initialization real elements from a dataloader are used.
Note that for export and for building NNCF graph (i.e. for dummy_forward function) some dummy random input is used -- the dummy input is generated once, its `img_meta` is also generated once, and the `img_meta` is stored in the model field.
For NNCF initialization real dataset elements are used with their `img_meta`-s. The `img_meta`-s may be different for different elements, so they cannot be stored as a field of the model.